src/courtroom.cpp

```diff
@@ -4465,6 +4476,77 @@ void Courtroom::refresh_iclog(bool skiplast)
   }
 }
 
+void Courtroom::truncate_label_text(QCheckBox* p_checkbox, QString p_identifier)

```

Comment:
Use QWidget for the argument and then use [`QObject::inherits`](https://doc.qt.io/qt-5/qobject.html#inherits) to see if `setTooltip` or `setText` are appropriate.

Reply:
I was unable to use QObject::inherits due to Qt Creator staunchly refusing to compile it, so as you advised on Discord I used qtobject_cast to condense the copy-pasted functions into one function. I kind of hate looking at this, but it works.